src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(action, value) {
+      action.call(this, value);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result = value === 0 ? 0 : this.result / value;
+    },
```

Comment:
The methods `add`, `subtract`, `multiply`, and `divide` are defined using the shorthand syntax which is good as it adheres to the [CODE KNOWLEDGE] checklist.